Mage.Sets/src/mage/cards/t/TurnTheTables.java

```diff
@@ -55,16 +55,15 @@ public TurnTheTablesEffect copy() {
     }
 
     @Override
-    protected PreventionEffectData preventDamageAction(GameEvent event, Ability source, Game game) {
-        PreventionEffectData preventionEffectData = super.preventDamageAction(event, source, game);
-        int damage = preventionEffectData.getPreventedDamage();
-        if (damage > 0) {
-            Permanent attackingCreature = game.getPermanent(getTargetPointer().getFirst(game, source));
-            if (attackingCreature != null) {
-                attackingCreature.damage(damage, source.getSourceId(), source, game, false, true);
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        DamageEvent damageEvent = (DamageEvent) event;
+        if (source.getControllerId() != null && game.getPermanent(source.getFirstTarget()) != null) {
```

Comment:
Thank you, I had indeed removed the crucial `game.getPlayer` when trimming the code. It's now there.

I'm not sure what you mean by A though, sorry.

Reply:
``Instead of

```
if (foo != nul) {
    ...
    return true
}
return false 
```

Try

```
if (foo == null) {
    return false
}
...
return true
```

Reply:
Thanks for the example, I get what you mean by the flattening now. Went and changed it to be like that.